packages/utils/src/mergeDefaultsWithFormData.ts

```diff
@@ -48,33 +48,43 @@ export default function mergeDefaultsWithFormData<T = any>(
           formData[idx],
           mergeExtraArrayDefaults,
           defaultSupercedesUndefined,
-          overrideFormDataWithDefaults,
+          overrideFormDataWithDefaultsStrategy,
         );
       }
       return value;
     });
 
     // Merge any extra defaults when mergeExtraArrayDefaults is true
-    // Or when overrideFormDataWithDefaults is true and the default array is shorter than the formData array
-    if ((mergeExtraArrayDefaults || overrideFormDataWithDefaults) && mapped.length < overrideOppositeArray.length) {
+    // Or when overrideFormDataWithDefaults is not noop and the default array is shorter than the formData array
```

Comment:
There is a difference between not `noop` as you've mentioned here and the `merge` value below. Consider updating comment if the code it correct or the code if the comment is correct